### PR TITLE
Function FQDN, fullHostName

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- New Feature

Short description (up to few sentences):
Add Function  FQDN aka fullHostName  #7263

### CODE REVIEW

File: dbms/src/Functions/FunctionFQDN.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    String getName() const override
+    {
+        return name;
+    }
+
+    bool isDeterministic() const override { return false; }
+
+    bool isDeterministicInScopeOfQuery() const override
+    {
+        return false;
```
Comment (alexey-milovidov) [inline dbms/src/Functions/FunctionFQDN.cpp]: Why not?

File: dbms/src/Functions/FunctionFQDN.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & /*arguments*/) const override
+    {
+        return std::make_shared<DataTypeString>();
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers &, size_t result, size_t input_rows_count) override
+    {
+        block.getByPosition(result).column = block.getByPosition(result).type->createColumnConst(
+            input_rows_count, Poco::Net::DNS::thisHost().name())->convertToFullColumnIfConst();
```
Comment (alexey-milovidov) [inline dbms/src/Functions/FunctionFQDN.cpp]: BTW, we have getFQDNOrHostName function that has two differences:
- it's cached;
- it has fallback to OS hostname if DNS does not work.

Comment (sundy-li) [reply dbms/src/Functions/FunctionFQDN.cpp]: Right, should it changes to the same behavior as getFQDNOrHostName() ？
That way the function should be named as `getFQDNOrHostName` too.

**Review by alexey-milovidov — APPROVED** [review]

File: dbms/src/Functions/FunctionFQDN.cpp (hunk 2 above)
Comment (alexey-milovidov) [reply dbms/src/Functions/FunctionFQDN.cpp]: > That way the function should be named as getFQDNOrHostName too.

I think, now it's Ok.

**Pull request merged**